freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/UserService.java

```diff
@@ -118,29 +115,28 @@ private void asyncSynchronizeUsers(
                                 return synchronizeStack(stack, envToUmsStateMap.get(stack.getEnvironmentCrn()), userIdFilter);
                             })));
 
-            List<SuccessDetails> success = new ArrayList<>();
-            List<FailureDetails> failure = new ArrayList<>();
-
             statusFutures.forEach((envCrn, statusFuture) -> {
```

Comment:
this doesn't really update the sync status on completion of each environment because statusFuture.get() is called serially. Update for completed environments can be blocked on the statusFuture.get() of an environment that it still running.
I think the status check and should be pushed into the synchronizeStack method.